src/state/nftMarket/helpers.ts

```diff
@@ -392,6 +428,53 @@ export const getNftsUpdatedMarketData = async (
   }
 }
 
+export const getAccountNftsOnChainMarketData = async (
+  collections: ApiCollections,
+  account: string,
+): Promise<TokenMarketData[]> => {
+  try {
+    const nftMarketAddress = getNftMarketAddress()
+    const collectionList = Object.values(collections)
+    const askCalls = collectionList.map((collection) => {
+      const { address: collectionAddress } = collection
+      return {
+        address: nftMarketAddress,
+        name: 'viewAsksByCollectionAndSeller',
+        params: [collectionAddress, account, 0, 1000],
+      }
+    })
+
+    const askCallsResultsRaw = await multicallv2(nftMarketAbi, askCalls, { requireSuccess: false })
+    const askCallsResults = askCallsResultsRaw
+      .map((askCallsResultRaw, askCallIndex) => {
+        if (!askCallsResultRaw.tokenIds || !askCallsResultRaw.askInfo) return null
+        return askCallsResultRaw.tokenIds
+          .map((tokenId, tokenIdIndex) => {
+            if (!tokenId || !askCallsResultRaw.askInfo[tokenIdIndex] || !askCallsResultRaw.askInfo[tokenIdIndex].price)
+              return null
+
+            const currentAskPrice = formatBigNumber(askCallsResultRaw.askInfo[tokenIdIndex].price)
+
+            return {
+              collection: { id: collectionList[askCallIndex].address.toLowerCase() },
```

Comment:
In any case, this `collectionList[askCallIndex]` returning null

Reply:
Done